tidy3d/components/geometry.py

```diff
@@ -1019,6 +1041,59 @@ def _arrow_dims(  # pylint: disable=too-many-locals
 
         return arrow_length, arrow_width
 
+    def volume(self, bounds: Bound) -> float:
+        """Returns object's volume. If the object extends beyound bounds then
+        the computed volume may be only a rough approximation
+
+        Returns
+        -------
+        float
+            Volume.
+        """
+        volume = 1
+
+        for axis in (0, 1, 2):
```

Comment:
```py
for axis in range(3):
```
is the standard way to do this in python.  Please replace also where this occurs in the methods below.